meson.build

```diff
@@ -0,0 +1,1852 @@
+substs.set('userpresetdir',                                   userpresetdir)
+substs.set('udevhwdbdir',                                     udevhwdbdir)
+substs.set('udevrulesdir',                                    udevrulesdir)
+substs.set('catalogdir',                                      catalogdir)
+substs.set('tmpfilesdir',                                     tmpfilesdir)
+substs.set('sysusersdir',                                     sysusersdir)
+substs.set('sysctldir',                                       sysctldir)
+substs.set('binfmtdir',                                       binfmtdir)
+substs.set('modulesloaddir',                                  modulesloaddir)
+substs.set('systemgeneratordir',                              systemgeneratordir)
+substs.set('usergeneratordir',                                usergeneratordir)
+substs.set('systemenvgeneratordir',                           systemenvgeneratordir)
+substs.set('userenvgeneratordir',                             userenvgeneratordir)
+substs.set('systemshutdowndir',                               systemshutdowndir)
+substs.set('systemsleepdir',                                  systemsleepdir)
+
+#####################################################################
+
+compiler = meson.get_compiler('c')
+pkgconfig = import('pkgconfig')
+
+foreach arg : ['-pipe',
+               '-Wall',
+               '-Wextra',
+               '-Wundef',
+               '-Wlogical-op',
+               '-Wmissing-include-dirs',
+               '-Wold-style-definition',
+               '-Wpointer-arith',
+               '-Winit-self',
+               '-Wdeclaration-after-statement',
+               '-Wfloat-equal',
+               '-Wsuggest-attribute=noreturn',
+               '-Werror=missing-prototypes',
+               '-Werror=implicit-function-declaration',
+               '-Werror=missing-declarations',
+               '-Werror=return-type',
+               '-Werror=incompatible-pointer-types',
+               '-Werror=format=2',
+               '-Wstrict-prototypes',
+               '-Wredundant-decls',
+               '-Wmissing-noreturn',
+               '-Wshadow',
+               '-Wendif-labels',
+               '-Wstrict-aliasing=2',
+               '-Wwrite-strings',
+               '-Wno-unused-parameter',
+               '-Wno-missing-field-initializers',
+               '-Wno-unused-result',
+               '-Wno-format-signedness',
+               '-Werror=overflow',
+               '-Wdate-time',
+               '-Wnested-externs',
+               '-ffast-math',
+               '-fno-common',
+               '-fdiagnostics-show-option',
+               '-fno-strict-aliasing',
+               '-fvisibility=hidden',
+               '-fstack-protector',
+               '-fstack-protector-strong',
+               '-fPIE',
+               '--param=ssp-buffer-size=4',
+              ]
+  if compiler.has_argument(arg)
+    add_global_arguments(arg, language : 'c')
```

Comment:
RIght. I read about the distinction between global and project, but it didn't really sink in. Sounds reasonable to move it though.